Src/EB/AMReX_EB_STL_utils.cpp

```diff
@@ -0,0 +1,151 @@
+    Print()<<"STL file name:"<<fname<<"\n";
+
+    std::getline(infile,tmpline); //solid <solidname>
+    while(!infile.eof())
+    {
+        std::getline(infile,tmpline);
+        if(tmpline.find("endsolid")!=std::string::npos)
+        {
+            break;
+        }
+        nlines++;
+    }
+
+    if(nlines%m_nlines_per_facet!=0)
+    {
+        Abort("may be there are blank lines in the STL file\n");
+    }
+
+    m_num_tri=nlines/m_nlines_per_facet;
+    Print()<<"number of triangles:"<<m_num_tri<<"\n";
+
+    //host vectors
+    m_tri_pts_h.resize(m_num_tri*m_ndata_per_tri);
+    m_tri_normals_h.resize(m_num_tri*m_ndata_per_normal);
+
+    infile.seekg(0);
+    std::getline(infile,tmpline); //solid <solidname>
+
+    for(int i=0;i<m_num_tri;i++)
+    {
+        std::getline(infile,tmpline);  //facet normal
+        std::istringstream fcnormal(tmpline);
+        fcnormal>>tmp1>>tmp2
+            >>m_tri_normals_h[i*m_ndata_per_normal+0]
+            >>m_tri_normals_h[i*m_ndata_per_normal+1]
+            >>m_tri_normals_h[i*m_ndata_per_normal+2];
+
+        std::getline(infile,tmpline); // outer loop
+
+        std::getline(infile,tmpline); //vertex 1
+        std::istringstream vertex1(tmpline);
+        vertex1>>tmp1
+            >>m_tri_pts_h[i*m_ndata_per_tri+0]
+            >>m_tri_pts_h[i*m_ndata_per_tri+1]
+            >>m_tri_pts_h[i*m_ndata_per_tri+2];
+
+        std::getline(infile,tmpline); //vertex 2
+        std::istringstream vertex2(tmpline);
+        vertex2>>tmp1 
+            >>m_tri_pts_h[i*m_ndata_per_tri+3]
+            >>m_tri_pts_h[i*m_ndata_per_tri+4]
+            >>m_tri_pts_h[i*m_ndata_per_tri+5];
+
+        std::getline(infile,tmpline); //vertex 3
+        std::istringstream vertex3(tmpline);
+        vertex3>>tmp1 //vertex
+            >>m_tri_pts_h[i*m_ndata_per_tri+6]
+            >>m_tri_pts_h[i*m_ndata_per_tri+7]
+            >>m_tri_pts_h[i*m_ndata_per_tri+8];
+
+        std::getline(infile,tmpline); //end loop
+        std::getline(infile,tmpline); //end facet
+    }
+  
+    //device vectors  
+    m_tri_pts_d.resize(m_num_tri*m_ndata_per_tri);
+    m_tri_normals_d.resize(m_num_tri*m_ndata_per_normal);
+
+    Gpu::copy(Gpu::hostToDevice, m_tri_pts_h.begin(),
+              m_tri_pts_h.end(), m_tri_pts_d.begin());
+    Gpu::copy(Gpu::hostToDevice, 
+              m_tri_normals_h.begin(), m_tri_normals_h.end(),
+              m_tri_normals_d.begin());
+}
+//================================================================================
+void STLtools::stl_to_markerfab(MultiFab& markerfab,Geometry geom,
+        Real *point_outside)
+{
+    //local variables for lambda capture
+    int data_stride   = m_ndata_per_tri;
+    int num_triangles = m_num_tri;
+    Real outvalue     = m_outside;
+    Real invalue      = m_inside;
+
+    const auto plo   = geom.ProbLoArray();
+    const auto dx    = geom.CellSizeArray();
+    Real outp[]      = {point_outside[0],point_outside[1],point_outside[2]};
```

Comment:
Can we make this `GpuArray<Real,3> outp`?

Reply:
I have changed this